tests/test_wallet.py

```diff
@@ -524,7 +524,7 @@ def test_txprepare_restart(node_factory, bitcoind, chainparams):
 
     # It should have logged this for each output.
     for i in decode['vin']:
-        assert l1.daemon.is_in_log('wallet: reserved output {}/{} reset to available'.format(i['txid'], i['vout']))
+        l1.daemon.wait_for_log('wallet: reserved output {}/{} reset to available'.format(i['txid'], i['vout']))
```

Comment:
This changes the requirement from loose ordering to strict ordering, which then promptly failed the test in CI.

I'd suggest the following instead:

```python3
template = r'wallet: reserved output {}/{} reset to available'
lines = [template.format(i['txid'], i['vout']) for i in decode['vin']]
l1.daemon.wait_for_logs(lines)
```

This will wait for the lines in any order.